raiwidgets/raiwidgets/_cohort.py

```diff
@@ -0,0 +1,186 @@
+                "Got unexpected type {0} for column. "
+                "Expected string type.".format(type(column))
+            )
+        if not isinstance(arg, list):
+            raise UserConfigValidationException(
+                "Got unexpected type {0} for arg. "
+                "Expected list type.".format(type(arg))
+            )
+        if len(arg) == 0:
+            raise UserConfigValidationException(
+                "Empty list supplied for arg."
+            )
+
+        if method in CohortFilterMethods.SINGLE_VALUE_METHODS:
+            if len(arg) != 1:
+                raise UserConfigValidationException(
+                    "Expected a single value in arg "
+                    "for cohort methods {0}.".format(
+                        " or ".join(CohortFilterMethods.SINGLE_VALUE_METHODS))
+                )
+            if not isinstance(arg[0], int) and not isinstance(arg[0], float):
+                raise UserConfigValidationException(
+                    "Expected int or float type for "
+                    "arg with cohort methods {0}.".format(
+                        " or ".join(CohortFilterMethods.SINGLE_VALUE_METHODS))
+                )
+
+        if method == CohortFilterMethods.InTheRangeOf:
+            if len(arg) != 2:
+                raise UserConfigValidationException(
+                    "Expected two entries in arg for "
+                    "cohort method {0}.".format(
+                        CohortFilterMethods.InTheRangeOf)
+                )
+            if ((not isinstance(arg[0], int) and
+                 not isinstance(arg[0], float)) or
+                (not isinstance(arg[1], int) and
+                 not isinstance(arg[1], float))):
+                raise UserConfigValidationException(
+                    "Expected int or float type for arg "
+                    "with cohort method {0}.".format(
+                        CohortFilterMethods.InTheRangeOf)
+                )
+
+
+class Cohort:
+    """Defines the cohort which will be injected from SDK into the Dashboard.
+
+    :param method: Name of the cohort.
+    :type method: str
+    """
+    def __init__(self, name: str):
+        """Defines the cohort which will be injected from SDK into the Dashboard.
+
+        :param method: Name of the cohort.
+        :type method: str
+        """
+        self.name = name
+        self.cohort_filter_list = None
+
+    def add_cohort_filter(self, cohort_filter: CohortFilter):
+        """Add a cohort filter into the cohort.
+
+        :param cohort_filter: Cohort filter defined by CohortFilter class.
+        :type: CohortFilter
+        """
+        if self.cohort_filter_list is None:
+            self.cohort_filter_list = [cohort_filter]
+        else:
+            self.cohort_filter_list.append(cohort_filter)
+
+    def _validate_cohort_with_test_data(
+            self, test_data: pd.DataFrame):
+        # TODO: Add validations for checking cohort and cohort filter against
```

Comment:
maybe remove this for now, since it's not implemented anyway?

Reply:
Removed it.